roles/openshift_health_checker/openshift_checks/logging/logging_index_time.py

```diff
@@ -0,0 +1,168 @@
+    """Check that pod logs are aggregated and indexed in ElasticSearch within a reasonable amount of time."""
+    name = "logging_index_time"
+    tags = ["health", "logging"]
+
+    logging_namespace = "logging"
+
+    def run(self, tmp, task_vars):
+        """Add log entry by making unique request to Kibana. Check for unique entry in the ElasticSearch pod logs."""
+        try:
+            log_index_timeout = int(
+                get_var(task_vars, "openshift_check_logging_index_timeout_seconds", default=ES_CMD_TIMEOUT_SECONDS)
+            )
+        except ValueError:
+            return {
+                "failed": True,
+                "msg": ('Invalid value provided in your Ansible inventory file for '
+                        '"openshift_check_logging_index_timeout_seconds". Value must '
+                        'be an integer representing an amount in seconds.'),
+            }
+
+        # get all Kibana pods
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        pods, error = super(LoggingIndexTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "kibana",
+            task_vars,
+        )
+
+        if error:
+            msg = 'Unable to retrieve pods for the "Kibana" logging component: {}'
+            return {"failed": True, "changed": False, "msg": msg.format(error)}
+
+        running_kibana_pods = self.running_pods(pods)
+        if not len(running_kibana_pods):
+            msg = ('No Kibana pods were found to be in the "Running" state.'
+                   'At least one Kibana pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        # get all Elasticsearch pods
+        pods, error = super(LoggingIndexTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "es",
+            task_vars,
+        )
+
+        if error:
+            msg = 'Unable to retrieve pods for the "Elasticsearch" logging component: {}'
+            return {"failed": True, "changed": False, "msg": msg.format(error)}
+
+        running_es_pods = self.running_pods(pods)
+        if not len(running_es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        uuid = self.curl_kibana_with_uuid(running_kibana_pods[0], task_vars)
+        self.wait_until_cmd_or_err(running_es_pods[0], uuid, log_index_timeout, task_vars)
+        return {}
+
+    def wait_until_cmd_or_err(self, es_pod, uuid, timeout_secs, task_vars):
+        """Retry an Elasticsearch query every second until query success, or a defined
+        length of time has passed."""
+        now = time.time()
```

Comment:
Original comment: https://github.com/openshift/openshift-ansible/pull/4519#discussion_r125238506

Let me try to explain again.

"Retry ... until a defined length of time has passed" sounds plausible, but is hard to implement without access to a monotonic time source. `time.time()` is not monotonic.

Instead of adding hacks to try to compensate for problems with `time.time()`, what I am proposing is an algorithm that doesn't depend on it: "retry N times with S seconds interval" (`time.sleep` uses a monotonic clock internally).

In code:

```python
interval = 1

for i in range(max_tries):
    if self.query_es_from_es(es_pod, uuid, task_vars):
        return
    if i < max_tries - 1:
        time.sleep(interval)  # note: we actually don't need to sleep after the last try...

msg = ...
raise ...
```